src/main/java/stringaddcalculator/PositiveNumberExtractor.java

```diff
@@ -0,0 +1,50 @@
+package stringaddcalculator;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class PositiveNumberExtractor {
+
+  private final static String DEFAULT_SEPARATOR = "[,:]";
+  private final static String CUSTOM_SEPARATOR_PATTERN = "//(.)\n(.*)";
+
+  public static List<PositiveNumber> create(String value) {
+    if (Objects.isNull(value) || value.isEmpty()) {
+      return Collections.singletonList(PositiveNumber.createForNullOrEmpty());
+    }
+    if (returnMatcher(DEFAULT_SEPARATOR, value).find()) {
+      return createByDefaultSeparator(value);
+    }
+    if (returnMatcher(CUSTOM_SEPARATOR_PATTERN, value).matches()) {
+      return createByCustomSeparator(value);
+    }
+    return Collections.singletonList(PositiveNumber.create(value));
+  }
+
+  public static List<PositiveNumber> createByDefaultSeparator(String value) {
+    return Arrays.stream(value.split(DEFAULT_SEPARATOR))
+        .map(PositiveNumber::create)
+        .collect(Collectors.toList());
+  }
+
+  public static List<PositiveNumber> createByCustomSeparator(String value) {
+    Matcher m = Pattern.compile(CUSTOM_SEPARATOR_PATTERN).matcher(value);
+    if (m.find()) {
+      String customDelimiter = m.group(1);
+      String[] tokens = m.group(2).split(customDelimiter);
+      return Arrays.stream(tokens)
+          .map(PositiveNumber::create)
+          .collect(Collectors.toList());
+    }
+    throw new RuntimeException("something is wrong.");
+  }
+
+  public static Matcher returnMatcher(String separator, String value) {
+    return Pattern.compile(separator).matcher(value);
```

Comment:
현재 매개변수 separator에 따라 패턴 매칭을 하여 메서드를 재사용할 수 있도록 의도하신 것 같아요!
다만 Pattern.compile() 내부 구현을 보면 
```
// Pattern
public static Pattern compile(String regex) {
        return new Pattern(regex, 0);
}
```
returnMatcher() 메서드를 호출할때마다 String을 패턴 매칭을 위해
매번 성능이 비싼 Pattern 인스턴스를 생성하고 Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않습니다. 
뿐더러 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다.

If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time

매번 불필요한 객체 생성을 피하도록 개선하면 좋을 것 같아요 :)

Reply:
네 동의합니다. 사실 해당 부분을 고려하지 않았던 것은 아니나 시간의 제약으로 적당히 물리고 넘어갔습니다. 컴파일한 클래스를 어떻게 해야 적절히 재사용할 수 있을까요?

Reply:
공유자원으로 메모리에 한번만 적재되도록 캐싱을 해보면 어떨까요 :) ?
```
private static final Pattern DEFAULT_PATTERN = Pattern.compile(DEFAULT_SEPARATOR);
```